src/interceptors/response.interceptor.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  Injectable,
+  NestInterceptor,
+  ExecutionContext,
+  CallHandler,
+  Logger,
+} from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+export interface Response<T> {
+  data: T;
+}
+
+@Injectable()
+export class ResponseInterceptor<T> implements NestInterceptor<T, Response<T>> {
+  private logger = new Logger();
+
+  intercept(
+    context: ExecutionContext,
+    next: CallHandler,
+  ): Observable<Response<T>> {
+    const statusCode = context.switchToHttp().getResponse().statusCode;
+    const req = context.switchToHttp().getRequest();
+    const userDetail = req?.user || {};
+
+    return next.handle().pipe(
+      map((data) => {
+        const response = {
+          isSuccess: statusCode >= 200 && statusCode < 300 ? true : false,
+          statusCode: statusCode,
+          error: data && data.message ? data.message : '',
+          message: data && data.message ? data.message : '',
+          data: data && data.message ? { ...data, message: undefined } : data,
+        };
+        this.logger.debug(
+          `Response: [${JSON.stringify(
+            response,
+          )}] || User Details: [${JSON.stringify(userDetail)}]`,
+        );
+
+        return response;
+      }),
+    );
```

Comment:
PrimathonAICodeReviewer

The logging and response creation logic could be extracted into separate functions for better modularity and readability.

```diff
+         const response = this.createResponse(data, statusCode);
+         this.logResponse(response, userDetail);
+
+         return response;
```

You can create the `createResponse` and `logResponse` functions within the `ResponseInterceptor` class to encapsulate these responsibilities.

```typescript
private createResponse(data: any, statusCode: number): Response<T> {
    return {
        isSuccess: statusCode >= 200 && statusCode < 300,
        statusCode: statusCode,
        error: data?.message || '',
        message: data?.message || '',
        data: data?.message ? { ...data, message: undefined } : data,
    };
}

private logResponse(response: Response<T>, userDetail: any) {
    this.logger.debug(
        `Response: [${JSON.stringify(response)}] || User Details: [${JSON.stringify(userDetail)}]`,
    );
}
```

### Overall, the changes look good with the suggested improvements. 



<!-- This is an auto-generated comment by OSS PrimathonAICodeReviewer -->